test/EventPublisher.test.js

```diff
@@ -9,302 +9,264 @@ import { getRequestEventName } from '../src/lib/EventPublisher';
-      },
-    );
-
-    eventPublisher.publish(deviceEvent);
-    eventPublisher.publish(notDeviceEvent);
-
-    process.nextTick(() => {
-      t.falsy(handler.called);
-    });
-  }
-);
-
-test(
-  'should filter broadcasted events',
-  t => {
-    const eventPublisher = new EventPublisher();
-    const handler = sinon.spy();
-    const ownerID = TestData.getID();
-    const deviceEvent = {
-      broadcasted: true,
-      deviceID: TestData.getID(),
-      isPublic: false,
-      name: TEST_EVENT_NAME,
-      userID: ownerID,
-    };
-
-    eventPublisher.subscribe(
-      deviceEvent.name,
-      handler,
-      {
-        filterOptions: {
-          listenToBroadcastedEvents: false,
-        },
-      },
-    );
-
-    eventPublisher.publish(deviceEvent);
-
-    process.nextTick(() => {
-      t.falsy(handler.called);
-    });
-  }
-);
-
-test(
-  'should listen for mydevices events only',
-  t => {
-    const eventPublisher = new EventPublisher();
-    const handler = sinon.spy();
-    const ownerID = TestData.getID();
-
-    const myDevicePublicEvent = {
-      name: TEST_EVENT_NAME,
-      userID: ownerID,
-      isPublic: true,
+test('should subscribe to event', t => {
+  const eventPublisher = new EventPublisher();
+  const handler = sinon.spy();
+  const eventData = {
+    name: TEST_EVENT_NAME,
+    userID: TestData.getID(),
+    isPublic: true,
+  };
+
+  eventPublisher.subscribe(eventData.name, handler, {
+    filterOptions: { userID: eventData.userID },
+  });
+
+  eventPublisher.publish(eventData);
+
+  process.nextTick(() => {
+    t.truthy(handler.called);
+  });
+});
+
+test('should listen for public event from another owner device', t => {
+  const eventPublisher = new EventPublisher();
+  const handler = sinon.spy();
+  const eventData = {
+    name: TEST_EVENT_NAME,
+    userID: TestData.getID(),
+    isPublic: true,
+  };
+
+  eventPublisher.subscribe(eventData.name, handler, {
+    filterOptions: { userID: TestData.getID() },
+  });
+
+  eventPublisher.publish(eventData);
+  process.nextTick(() => {
+    t.truthy(handler.called);
+  });
+});
+
+test('should filter private event', t => {
+  const eventPublisher = new EventPublisher();
+  const handler = sinon.spy();
+  const eventData = {
+    name: TEST_EVENT_NAME,
+    userID: TestData.getID(),
+    isPublic: false,
+  };
+
+  eventPublisher.subscribe(eventData.name, handler, {
+    filterOptions: { userID: TestData.getID() },
+  });
+
+  eventPublisher.publish(eventData);
+
+  process.nextTick(() => {
+    t.falsy(handler.called);
+  });
+});
+
+test('should filter internal event', t => {
+  const eventPublisher = new EventPublisher();
+  const handler = sinon.spy();
+  const eventData = {
+    name: TEST_EVENT_NAME,
+    isInternal: true,
+  };
+
+  eventPublisher.subscribe(eventData.name, handler, {
+    filterOptions: { listenToInternalEvents: true },
+  });
+
+  eventPublisher.publish(eventData);
+
+  process.nextTick(() => {
+    t.falsy(handler.called);
```

Comment:
Hey.. You should do multiple PRs instead of making a giant PR.

If the tests should be rewritten, it should be done in separate PR so it's easier to review.